metriq_gym/benchmarks/quantum_volume.py

```diff
@@ -1,45 +1,219 @@
+    numer: float = 0
+    denom: float = 0
+    sum_hog_counts = 0
+    for i in range(n_pow):
+        b = (bin(i)[2:]).zfill(n)
+
+        count = counts[b] if b in counts else 0
+        ideal = ideal_probs[i]
+
+        # XEB / EPLG.
+        denom = denom + (ideal - u_u) ** 2
+        numer = numer + (ideal - u_u) * ((count / shots) - u_u)
+
+        # QV / HOG.
+        if ideal > threshold:
+            sum_hog_counts += count
+
+    hog_prob = sum_hog_counts / shots
+    xeb = numer / denom if denom > 0 else 0
+    p_val = (1 - binom.cdf(sum_hog_counts - 1, shots, 1 / 2).item()) if sum_hog_counts > 0 else 1
+
+    return TrialStats(
+        qubits=n,
+        shots=shots,
+        xeb=xeb,
+        hog_prob=hog_prob,
+        hog_pass=hog_prob >= 2 / 3,
+        p_value=p_val,
+        confidence_level=confidence_level,
+        confidence_pass=p_val < confidence_level,
+        eplg=(1 - (xeb ** (1 / n))) if xeb < 1 else 0,
+    )
+
+
+def calc_stats(data: QuantumVolumeData, counts: list[MeasCount]) -> AggregateStats:
+    """Calculate aggregate statistics over multiple trials.
+
+    Args:
+        data: contains dispatch-time data (input data + ideal probability).
+        counts: contains results from the quantum device (one MeasCount per trial).
+    Returns:
+        A list of `AggregateStats` objects, each containing aggregated statistics for a result.
+    """
+    trial_stats = []
+
+    num_trials = len(counts)
+    # Process each trial, handling provider-specific logic.
+    for trial in range(num_trials):
+        trial_stat = calc_trial_stats(
+            ideal_probs=data.ideal_probs[trial],
+            counts=counts[trial],
+            shots=data.shots,
+            confidence_level=data.confidence_level,
         )
+        trial_stats.append(trial_stat)
+
+    # Aggregate the trial statistics.
+    hog_prob = sum(stat.hog_prob for stat in trial_stats) / num_trials
+    p_value = math.prod(stat.p_value for stat in trial_stats) ** (1 / num_trials)
 
-        if len(result.counts) > 0:
-            stats = calc_stats([result])
-            logging.info(f"Simulator-only job completed: {stats[0]}.")
-            print(stats[0])
-            return
+    return AggregateStats(
+        trials=num_trials,
+        trial_p_values=[stat.p_value for stat in trial_stats],
+        hog_prob=hog_prob,
+        p_value=p_value,
+        hog_pass=all(stat.hog_pass for stat in trial_stats),
+        confidence_pass=all(stat.confidence_pass for stat in trial_stats),
+    )
 
-        logging.info(f"Dispatched {self.params["trials"]} trials in 1 job.")
 
-        self.job_manager.add_job(result.to_serializable())
+class QuantumVolume(Benchmark):
+    def dispatch_handler(self, device: QuantumDevice) -> QuantumVolumeData:
+        num_qubits = self.params.num_qubits
+        shots = self.params.shots
+        trials = self.params.trials
+        circuits, ideal_probs = prepare_qv_circuits(num_qubits, trials)
```

Comment:
[nit]: Like you do for the call below, can we explicitly pass the arguments to the function by name (just to ensure that the order or arguments is okay and clear).